### DISCUSSION TITLE
Upgrade paths for OKD and commercial OpenShift are not the same?

### DISCUSSION THREAD

**Discussion by doman18** [okd-project/okd#1970] [Q&A] [answered]
Redhat offers this upgrade planning tool for openshift. 

https://access.redhat.com/labs/ocpupgradegraph/update_path/

And it works perfectly fine with openshift (ive tested it with simple SNO cluster and "stable" channels). I tried to use it for OKD (stable channels) but actual available versions are differend in OKD console

For example i wanted to upgrade okd 4.11.0.okd-2022-12-02-145640 to 4.15.x. In provided tool i could not choose used channel (stable-4) so i chose the closest one (stable-4.11). Graph returned this path

4.11.0 -> 4.11.17 -> 4.12.50 -> 4.13.34 -> 4.14.14 -> 4.15.0

And from the beginning the path was wrong because update installer would not allow me to choose provided versions. I only tried 4.11 to 4.12 upgrade for now. And real upgrade path was like this

4.11.0-okd-2022-12-02-145640   
4.11.0-okd-2023-01-14-152430
4.12.0-okd-2023-04-16-041331

I also asked ChatGPT "what is the upgrade path for okd from 4.8.0-0.okd-2021-10-24-061736 to 4.15 stable" (because i have also 2 clusters with 4.8 version) and at first it also showed the same path as upgrade planning tool showed. But i challenged it, i showed it that path is different and later it suggested to use the lastest version from particular channel

So overall is there a way to build upgrade plan (similar to provided for commercial openshift) for OKD? Clients of company i work for keep asking for them when we do upgrades for them.

**Comment by JaimeMagiera** [comment]
Thank you for your question regarding a way to map upgrade paths. We currently do not have such a tool. I'm not sure if you saw our [recent announcement](https://www.okd.io/blog/2024/06/01/okd_future_statement/). One thing we can say definitively is that you'll need to upgrade clusters from 4.14 FCOS to either 4.15 SCOS or 4.16 SCOS. We're working out the details of what the upgrade graph would look like from that point on. 

One thing I would point out is that the community builds and provides resources only for n and n-1 (e.g. 4.15 and 4.16). Moving forward, it would be beneficial to keep clusters on the latest minor or previous minor versions. While we try our best to help, we only have so many resources to do so. 

Does that help lay things out? Any other questions?

**Comment by Goose29** [comment, marked as answer]
@doman18 Although it is not "official" I have created a tool for this some time ago and contributed it. 

It can be found in the [FAQ docs](https://github.com/okd-project/okd/blob/master/FAQ.md#how-do-i-find-a-valid-upgrade-path).

I still use it to find valid upgrade paths.

**Comment by doman18** [comment]
@Goose29 
I know its been 2 weeks since you replied but i want to THANK YOU for link to the tool. This is EXACTLY what i needed! I've just successfully upgraded my testing cluster from 4.8.0-0.okd-2021-10-24-061736 to 4.15.0-0.okd-2024-03-10-010116 following this tree

https://amd64.origin.releases.ci.openshift.org/graph?format=png

@JaimeMagiera 
> One thing we can say definitively is that you'll need to upgrade clusters from 4.14 FCOS to either 4.15 SCOS or 4.16 SCOS.

Are you sure you ment 'from 4.14 FCOS' and not 'from 4.15'. Ive just upgraded cluster to 4.15, will i be able to update it to SCOS?

**Reply by JaimeMagiera** [reply]
Sorry, yes. It's been a long past couple of months.
